core/src/main/java/com/datastrato/gravitino/storage/kv/BinaryEntityEncoderUtil.java

```diff
@@ -0,0 +1,185 @@
+   * and the name is 'schema', the result will be '1/2/sc_schema'.
+   *
+   * <p>Attention, in order to make this change backward compatible, if the entity type is TABLE, we
+   * will not add a prefix to the name. If the entity type is not TABLE, we will add the prefix to
+   * the name.
+   *
+   * @param namespaceIds namespace ids, which are the ids of the parent entities
+   * @param name name of the entity
+   * @param type type of the entity
+   * @return concatenated string that used in the id-name mapping.
+   */
+  public static String concatIdAndName(long[] namespaceIds, String name, EntityType type) {
+    String context =
+        Joiner.on(NAMESPACE_SEPARATOR)
+            .join(
+                Arrays.stream(namespaceIds).mapToObj(String::valueOf).collect(Collectors.toList()));
+    // We need to make it backward compatible, so we need to check if the name is already prefixed.
+    String mappingName =
+        VERSION_0_4_0_COMPATIBLE_ENTITY_TYPES.contains(type)
+            ? name
+            : type.getShortName() + TYPE_AND_NAME_SEPARATOR + name;
+    return StringUtils.isBlank(context) ? mappingName : context + NAMESPACE_SEPARATOR + mappingName;
+  }
+
+  /**
+   * Get key prefix of all sub-entities under a specific entities. For example, as a metalake will
+   * start with `ml/{metalake_id}`, sub-entities under this metalake will have the prefix
+   *
+   * <pre>
+   *   catalog: ca/{metalake_id}
+   *   schema:  sc/{metalake_id}
+   *   table:   ta/{metalake_id}
+   * </pre>
+   *
+   * Why the sub-entities under this metalake start with those prefixes, please see {@link
+   * KvEntityStore} java class doc.
+   *
+   * @param ident identifier of an entity
+   * @param type type of entity
+   * @return list of sub-entities prefix
+   * @throws IOException if error occurs
+   */
+  public static List<byte[]> getSubEntitiesPrefix(
+      NameIdentifier ident, EntityType type, BinaryEntityKeyEncoder entityKeyEncoder)
+      throws IOException {
+    List<byte[]> prefixes = Lists.newArrayList();
+    byte[] encode = entityKeyEncoder.encode(ident, type, true);
+    switch (type) {
+      case METALAKE:
+        prefixes.add(replacePrefixTypeInfo(encode, CATALOG.getShortName()));
+        prefixes.add(replacePrefixTypeInfo(encode, SCHEMA.getShortName()));
+        prefixes.add(replacePrefixTypeInfo(encode, TABLE.getShortName()));
+        prefixes.add(replacePrefixTypeInfo(encode, FILESET.getShortName()));
+        break;
+      case CATALOG:
+        prefixes.add(replacePrefixTypeInfo(encode, SCHEMA.getShortName()));
+        prefixes.add(replacePrefixTypeInfo(encode, TABLE.getShortName()));
+        prefixes.add(replacePrefixTypeInfo(encode, FILESET.getShortName()));
+        break;
+      case SCHEMA:
+        prefixes.add(replacePrefixTypeInfo(encode, TABLE.getShortName()));
+        prefixes.add(replacePrefixTypeInfo(encode, FILESET.getShortName()));
+        break;
+      case TABLE:
+      case FILESET:
```

Comment:
Should you also add USER and GROUP here?

Reply:
If we add USER and GROUP here, USER and GROUP will block the non-cascade deletion of metalake.